sklearn/pipeline.py

```diff
@@ -1226,6 +1226,15 @@ def __sklearn_tags__(self):
                 tags.input_tags.pairwise = get_tags(
                     self.steps[0][1]
                 ).input_tags.pairwise
+            # WARNING: the sparse tag can be incorrect.
+            # Some Pipelines accepting sparse data are wrongly tagged sparse=False.
+            # For example Pipeline([PCA(), estimator]) accepts sparse data
+            # even if the estimator doesn't as PCA outputs a dense array.
+            tags.input_tags.sparse = all(
+                get_tags(step).input_tags.sparse
+                for name, step in self.steps
+                if step != "passthrough"
+            )
```

Comment:
This makes me wonder if we aren't expecting too much from the sparse tag. Maybe we should narrow its scope to the estimator level, regardless of whether or not its sub-estimator(s), if any, do support sparse input. In that case sparse means "this estimator can handle sparse input but its inner estimator(s) (if it's a meta-estimator) might not in which case it's undefined behavior". What do you think @glemaitre and @adrinjalali ?

Reply:
As discussed irl with Guillaume, an other option is to change `sparse` to accept ``True``, ``False`` and ``Maybe`` (or ``Undefined``, name tbd).